components/filter/filter.js

```diff
@@ -710,6 +732,24 @@ class Filter extends FocusMixin(LocalizeCoreElement(RtlMixin(LitElement))) {
 		this._activeDimensionKey = null;
 	}
 
+	_handleDimensionLoadMore(e) {
+		const dimensionKey = e.target.parentNode.id.slice(SET_DIMENSION_ID_PREFIX.length);
+		const dimension = this._getDimensionByKey(dimensionKey);
+		const searchCompleteCallback = this._getSearchCallback(dimension);
+
+		this.dispatchEvent(new CustomEvent('d2l-filter-dimension-load-more', {
+			detail: {
+				dimensionKey,
+				complete: (keysToDisplay) => {
+					searchCompleteCallback({ keysToDisplay });
```

Comment:
Still trying to wrap my head around how this works if the consumer can't give us all the values up front and will need to go fetch them when they get this event, which I _expect_ is what everyone using `load-more` will need for performance reasons.

I think this will happen, which won't work:
1. We fire this event, and the consumer loads the new values on their side and adds new `d2l-filter-dimension-set-value`s for them.
2. Then they call `complete` and tell us what keys to display (they will probably need to be given the search value with the load event to know this?)
3. But step 1 triggers `_handleSlotChange`, which I think will clear everything related to searching

Maybe we book a meeting for when @geurts is back, because I think we need answers to these to figure out how to set this up: 
- What happens when someone clicks load more when a search is applied - does it give _only_ the new values that match that search? Does it load 20 more values but only show ones that match, meaning you could get no new items added?
- When you remove that search, does it keep the new values in the list, but then there's "missing" values in between?

It would be easiest to just disable load more if you've searched something, but I think that's the exact use case nova needs.